catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/security/accesscontrol/AccessControlPreQueryPluginTest.java

```diff
@@ -0,0 +1,320 @@
+                            .is()
+                            .equalTo()
+                            .text(EMAIL_BOB),
+                        FILTER_BUILDER
+                            .attribute(Security.ACCESS_ADMINISTRATORS)
+                            .is()
+                            .equalTo()
+                            .text(EMAIL_BOB),
+                        FILTER_BUILDER
+                            .attribute(Security.ACCESS_GROUPS)
+                            .is()
+                            .equalTo()
+                            .text(ROLE_A),
+                        FILTER_BUILDER
+                            .attribute(Security.ACCESS_GROUPS_READ)
+                            .is()
+                            .equalTo()
+                            .text(ROLE_A),
+                        FILTER_BUILDER
+                            .attribute(Security.ACCESS_GROUPS)
+                            .is()
+                            .equalTo()
+                            .text(ROLE_B),
+                        FILTER_BUILDER
+                            .attribute(Security.ACCESS_GROUPS_READ)
+                            .is()
+                            .equalTo()
+                            .text(ROLE_B))))));
+  }
+
+  private void verifyPluginDoesNotAlterTheFilter(Filter inputFilter)
+      throws PluginExecutionException, StopProcessingException {
+    QueryRequest request = plugin.process(new QueryRequestImpl(new QueryImpl(inputFilter)));
+    assertThat(asGeotoolsComparableFilter(request.getQuery()), is(equalTo(inputFilter)));
+  }
+
+  private static Filter asGeotoolsComparableFilter(Query query) {
+    return (Filter) query.accept(COPYING_VISITOR, null);
+  }
+
+  private static class AccessControlPreQueryPluginUnderTest extends AccessControlPreQueryPlugin {
+    private final List<String> subjectRoles;
+
+    private AccessControlPreQueryPluginUnderTest(
+        FilterBuilder filterBuilder, String... subjectRoles) {
+      super(
+          filterBuilder,
+          null /* Only method interacting with this ref gets overridden */,
+          config(""));
+      this.subjectRoles = Arrays.asList(subjectRoles);
+      this.init();
+    }
+
+    private AccessControlPreQueryPluginUnderTest(
+        FilterBuilder filterBuilder, List<String> subjectRoles, String groupThatCanSeeEverything) {
+      super(filterBuilder, null, config(groupThatCanSeeEverything));
+      this.subjectRoles = subjectRoles;
+      this.init();
+    }
+
+    @Override
+    String getSubjectIdentifier() {
+      return EMAIL_BOB;
+    }
+
+    @Override
+    List<String> getSubjectGroups() {
+      return subjectRoles;
+    }
+
+    private void init() {
+      ACCESS_CONTROLLED_TAGS.forEach(tag -> this.bindTag(refForTag(tag)));
+    }
+
+    private void unbind(String tag) {
+      this.unbindTag(refForTag(tag));
+    }
+
+    private static AccessControlSecurityConfiguration config(String groupThatCanSeeEverything) {
```

Comment:
❓ Any reason not to mock the `AccessControlSecurityConfiguration` in `setUp()`?

Reply:
No reason to mock data objects. Mocking something should be justified, not the other way around. 
https://github.com/mockito/mockito/wiki/How-to-write-good-tests#dont-mock-value-objects